thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/MetricCorrelationRankingPipeline.java

```diff
@@ -0,0 +1,348 @@
+    requestList.addAll(makeRequests(candidateMetrics, baselineRange, PRE_BASELINE));
+
+    LOG.info("Requesting {} time series", requestList.size());
+
+    Map<String, RequestContainer> requests = new HashMap<>();
+    for(RequestContainer c : requestList)
+      requests.put(c.request.getRequestReference(), c);
+
+    // fetch responses and calculate derived metrics
+    Map<String, DataFrame> responses = new HashMap<>();
+    try {
+      List<ThirdEyeRequest> thirdeyeRequests = new ArrayList<>();
+      for(RequestContainer c : requestList)
+        thirdeyeRequests.add(c.request);
+
+      Collection<ThirdEyeResponse> result = this.cache.getQueryResultsAsyncAndWait(thirdeyeRequests).values();
+
+      for(ThirdEyeResponse r : result) {
+        String id = r.getRequest().getRequestReference();
+        DataFrame df = parseResponse(r);
+        evaluateExpressions(df, requests.get(id).expressions);
+        if(LOG.isDebugEnabled())
+          LOG.debug("DataFrame '{}':\n{}", id, df);
+
+        responses.put(id, df);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    // determine current-baseline changes
+    Map<MetricEntity, DoubleSeries> pctChanges = new HashMap<>();
+    for(MetricEntity e : allMetrics) {
+      String currentId = makeIdentifier(e.getUrn(), PRE_CURRENT);
+      String baselineId = makeIdentifier(e.getUrn(), PRE_BASELINE);
+
+      if(!responses.containsKey(currentId)) {
+        LOG.warn("No current data for '{}'. Skipping.", e.getUrn());
+        continue;
+      }
+
+      if(!responses.containsKey(baselineId)) {
+        LOG.warn("No baseline data for '{}'. Skipping.", e.getUrn());
+        continue;
+      }
+
+      DataFrame current = responses.get(currentId);
+      DataFrame baseline = responses.get(baselineId);
+
+      DoubleSeries pctChange = current.joinInner(baseline).map(new Series.DoubleFunction() {
+        @Override
+        public double apply(double... values) {
+          return (values[0] - values[1]) / values[1];
+        }
+      }, COL_VALUE, COL_VALUE + "_right");
+
+      pctChanges.put(e, pctChange);
+    }
+
+    // determine relevance - by strength of correlation
+    Map<MetricEntity, Double> scores = new HashMap<>();
+    for(MetricEntity target : targetMetrics) {
+      if(!pctChanges.containsKey(target)) {
+        LOG.warn("No diff data for target metric '{}'. Skipping.", target.getUrn());
+        continue;
+      }
+
+      DoubleSeries changesTarget = pctChanges.get(target);
+
+      for(MetricEntity candidate : candidateMetrics) {
+        if(!pctChanges.containsKey(target)) {
+          LOG.warn("No diff data for candidate metric '{}'. Skipping.", candidate.getUrn());
+          continue;
+        }
+
+        DoubleSeries changesCandidate = pctChanges.get(candidate);
+
+        try {
+          double score = Math.abs(changesTarget.corr(changesCandidate)) * target.getScore();
```

Comment:
the current weighting/importance of the target metric. when the user's search context contains multiple metrics (with potentially different weightings)